File: lab5-demos/retrofit-demo-java/app/src/main/java/com/smd/retrofitdemo/network/PeopleInSpaceProvider.java
package com.smd.retrofitdemo.network;

import com.facebook.stetho.okhttp3.StethoInterceptor;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Provides the retrofit service for calling the ISS API
 */
public class PeopleInSpaceProvider {
    private static final String BASE_URL = "http://api.open-notify.org";

    private static Retrofit retrofit = null;

    private PeopleInSpaceProvider() {
    }

    public static PeopleInSpaceService getApiService() {
        if (retrofit == null) {
            retrofit = new Retrofit
                    .Builder()
                    .baseUrl(BASE_URL)
                    .client(new OkHttpClient.Builder()
                            .addNetworkInterceptor(new StethoInterceptor())
                            .build())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit.create(PeopleInSpaceService.class);
    }
}

File: lab5-demos/retrofit-demo-kotlin/app/src/main/java/com/smd/retrofitdemo/network/PeopleInSpaceProvider.kt
package com.smd.retrofitdemo.network

import com.squareup.moshi.Moshi
import retrofit2.Retrofit
import retrofit2.converter.moshi.MoshiConverterFactory

/**
 * Provides the retrofit service for calling the ISS API
 */
object PeopleInSpaceProvider {
    private const val BASE_URL = "http://api.open-notify.org"

    private val moshi = Moshi.Builder().build()

    private val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(MoshiConverterFactory.create(moshi))
        .build()

    val apiService: PeopleInSpaceService by lazy {
        retrofit.create(PeopleInSpaceService::class.java)
    }
}